plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointWriter.java

```diff
@@ -122,22 +119,20 @@ public void write(ConnectorSession session, CheckpointEntries entries, Path targ
                 addEntryType,
                 removeEntryType);
 
-        Properties schema = buildHiveSchema(columnNames, columnTypes);
-
-        Configuration conf = hdfsEnvironment.getConfiguration(new HdfsContext(session), targetPath);
-        configureCompression(conf, SNAPPY);
-        JobConf jobConf = toJobConf(conf);
-
-        RecordFileWriter writer = new RecordFileWriter(
-                targetPath,
-                columnNames,
-                fromHiveStorageFormat(PARQUET),
-                schema,
-                PARQUET.getEstimatedWriterMemoryUsage(),
-                jobConf,
-                typeManager,
-                DateTimeZone.UTC,
-                session);
+        ParquetSchemaConverter schemaConverter = new ParquetSchemaConverter(columnTypes, columnNames, false, false);
+
+        ParquetWriter writer = new ParquetWriter(
+                fileSystemFactory.create(session)
+                        .newOutputFile(targetPath)
+                        .create(newSimpleAggregatedMemoryContext()),
```

Comment:
Didn't you make memory context optional in another PR?

Reply:
You're right, this is in the ACID PR. I'll merge that first.